drivers/spi/spi_mcux_lpspi.c

```diff
@@ -322,26 +322,64 @@ static void spi_mcux_dma_callback(const struct device *dev, void *arg,
 		}
 	}
 #if CONFIG_SPI_ASYNC
+	int ret;
+
 	if (data->ctx.asynchronous &&
 	((data->status_flags & SPI_MCUX_LPSPI_DMA_DONE_FLAG)  ==
 	SPI_MCUX_LPSPI_DMA_DONE_FLAG)) {
+		/* dma_size of completed transfers were the same for TX and RX */
+		size_t dma_size = data->ctx.tx_len;
+
+		spi_context_update_tx(&data->ctx, 1, dma_size);
+		spi_context_update_rx(&data->ctx, 1, dma_size);
+
+		if (data->ctx.tx_len == 0 && data->ctx.rx_len == 0) {
+			spi_context_complete(&data->ctx, spi_dev, 0);
+			return;
+		}
+
+		/* Restart DMA for next buffer */
 		/* Load dma blocks of equal length */
-		size_t dma_size = MIN(data->ctx.tx_len, data->ctx.rx_len);
+		dma_size = MIN(data->ctx.tx_len, data->ctx.rx_len);
 
 		if (dma_size == 0) {
 			dma_size = MAX(data->ctx.tx_len, data->ctx.rx_len);
 		}
 
-		spi_context_update_tx(&data->ctx, 1, dma_size);
-		spi_context_update_rx(&data->ctx, 1, dma_size);
+		/* Clear status flags */
+		data->status_flags = 0U;
 
-		if (data->ctx.tx_len == 0 && data->ctx.rx_len == 0) {
-			spi_context_complete(&data->ctx, spi_dev, 0);
+		/* Restart the RX DMA */
+		ret = dma_reload(dev,
+				 data->dma_rx.channel,
+				 data->dma_rx.dma_blk_cfg.source_address,
+				 (uint32_t) data->ctx.rx_buf,
+				 dma_size);
+		if (ret != 0) {
+			LOG_ERR("RX dma_reload() failed with error %x.", ret);
+			data->status_flags |= SPI_MCUX_LPSPI_DMA_ERROR_FLAG;
+		}
+
+		/* Restart the TX DMA */
+		ret = dma_reload(dev,
```

Comment:
Have we verified this works correctly when the SPI is only sending or only receiving data? If memory serves, we usually configure the TX DMA or RX DMA with a step size of zero and a dummy buffer in these cases, so that DMA will run on both channels. My concern is that if we (for example) are sending TX data and not receiving anything, the LPSPI RX FIFO will overflow since the DMA will not be reading from it.